synapse/handlers/federation.py

```diff
@@ -1617,7 +1617,7 @@ async def _resume_sync_partial_state_room(self) -> None:
     async def _sync_partial_state_room(
         self,
         initial_destination: Optional[str],
-        other_destinations: Collection[str],
+        other_destinations: Sequence[str],
```

Comment:
can you explain why this is needed? `_prioritise_destinations_for_partial_state_resync` takes a `Collection`, not a `Sequence`.

Reply:
I think this might have been a left-over from a WIP version before I pulled out the helper. (Or maybe I was trying to express "we preserve the iteration order of `other_destinations`"? :shrug: )

Reply:
1ccf9b8063, in any case